### PR TITLE
[staging] removed full from vacuum and added "closing-time" to awards backfill

### PR DESCRIPTION
**Pull request by tony-sappe**
### Update `recompute_all_awards.py` script [DEV-3442] 
**Description:**
- Removed full from the vacuum statement so the table isn't prohibitively locked.
- Added "closing-time" cli arg to set the desired script end datetime

### CODE REVIEW

File: usaspending_api/database_scripts/job_archive/recompute_all_awards.py [hunk 1]
```diff
@@ -309,11 +333,27 @@ def main():
...
-            with Timer("[Awards {:,} - {:,}]".format(batch_min, batch_max), pipe_output=logging.info):
+            if CLOSING_TIME:
+                if ITERATION_ESTIMATED_SECONDS:
+                    curr_time = datetime.now(timezone.utc)
+                    next_run_estimated_end_datetime = curr_time + timedelta(seconds=ITERATION_ESTIMATED_SECONDS)
+                    dt_str = next_run_estimated_end_datetime.isoformat()
+                    if next_run_estimated_end_datetime >= CLOSING_TIME:
+                        logging.info("=> Estimated loop end datetime of: {}".format(dt_str))
+                        logging.info("=> Next IDs to process: {} - {}".format(batch_min, batch_max))
+                        logging.info("=> You don't have to go home, but you can't... stay...... heeeeere")
```
Comment (BrianZito) [inline usaspending_api/database_scripts/job_archive/recompute_all_awards.py]:
```suggestion
                        logging.info("=> Suspending procedure around closing time of {}".format(CLOSING_TIME))
                        logging.info("=> You don't have to go home, but you can't... stay...... heeeeere")
```
I know they would have to add the flag themselves, but future devs (or OPs trying to read logs to audit something) might not appreciate our humor at 2 in the morning why trying to figure out when things stopped running.

File: usaspending_api/database_scripts/job_archive/recompute_all_awards.py [hunk 2]
```diff
@@ -309,11 +333,25 @@ def main():
...
-            with Timer("[Awards {:,} - {:,}]".format(batch_min, batch_max), pipe_output=logging.info):
+            if CLOSING_TIME:
+                if ITERATION_ESTIMATED_SECONDS:
+                    curr_time = datetime.now(timezone.utc)
+                    next_run_estimated_end_datetime = curr_time + timedelta(seconds=ITERATION_ESTIMATED_SECONDS)
+                    dt_str = next_run_estimated_end_datetime.isoformat()
+                    logging.info("=> Expected interation duration: {} ".format(ITERATION_ESTIMATED_SECONDS))
+                    logging.info("=> Estimated loop end datetime of: {}".format(dt_str))
+                    if next_run_estimated_end_datetime >= CLOSING_TIME:
+                        logging.info("===== Start next job at ID {} =====".format(batch_min))
```
Comment (BrianZito) [inline usaspending_api/database_scripts/job_archive/recompute_all_awards.py]:
```suggestion
                        logging.info("===== Suspending job due to --closing-time flag. Start next job at ID {} =====".format(batch_min))
```

Comment (tony-sappe) [reply usaspending_api/database_scripts/job_archive/recompute_all_awards.py]: Proposed edits are too long. See pushed change

File: usaspending_api/database_scripts/job_archive/recompute_all_awards.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/database_scripts/job_archive/recompute_all_awards.py]: The irony of your concern is that this log message would **only** be printed if the script stopped _when it was set to stop_

Comment (BrianZito) [reply usaspending_api/database_scripts/job_archive/recompute_all_awards.py]: but it doesn't explain why. If I'm looking at the last ten runs for some reason, I'm left guessing why it decided to print this and stop all the sudden. I'll be able to control-f the code real quick for this very distinct string, which is why I don't think it's an issue, but we could help an ops guy by being more explicit.

**Comment by tony-sappe** [comment]
Travis-CI checks fails because of fiscal year tests. While fixed by this PR #2060 the edits have not reached staging yet

**Pull request merged**